docs/current_docs/api/constructors.mdx

```diff
@@ -112,6 +99,10 @@ If you plan to use constructor fields in other module functions, ensure that the
 
 Constructors can be passed both simple and complex types (such as `Container`, `Directory`, `Service` etc.) as arguments. Default values can be assigned in both cases.
 
+:::important
+Explicitly declare the type even when a default value is assigned, so that the Dagger SDK can extend the GraphQL schema correctly.
```

Comment:
This may not make sense in all languages. In Go, for example, it won't compile if you don't and default values aren't even supported in the language, we do it via a specially formatted comment.

This is more general:

```suggestion
Explicitly declare the type even when it can be inferred, so that the Dagger SDK can extend the GraphQL schema correctly.
```

Even then it's possible for an SDK to actually support type inferrence when registering types to the API, depending on the language.